### PR TITLE
Task/graph v2

### PR DESCRIPTION
**Pull request by alonisser**

### CODE REVIEW

File: uncontrast_studies/processors/experiments_comparison.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+            .values("name")
+            .annotate(experiment_count=Count("target_stimuli__experiment", distinct=True))
+            .annotate(key=F("name"))
+        )
+
+        return subquery
+
+    def process_consciousness_measure_phase(self, experiments: QuerySet[UnConExperiment]):
+        subquery = (
+            UnConsciousnessMeasurePhase.objects.filter(unconsciousness_measures__experiment__in=experiments)
```
Comment (dani-efr) [inline uncontrast_studies/processors/experiments_comparison.py:111]: בטוח שזה unconsciousness_measures ולא unconsciousness_measure? וגם בכל המקומות בהמשך

File: uncontrast_studies/processors/experiments_comparison.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    def aggregate_query_by_breakdown(self):
+        significance_options = SignificanceChoices.values
+
+        results = []
+        experiment_ids = []
+        for sig_option in significance_options:
+            experiments_for_option = (
+                self.experiments.filter(significance=sig_option).distinct().values_list("id", flat=True)
+            )
+            # Children_experiments is referring from theory to child theory and from their to "experiments"
```
Comment (dani-efr) [inline uncontrast_studies/processors/experiments_comparison.py:202]: what?
מה הקשר ל theory?

File: uncontrast_studies/processors/parameters_distribution_free_queries.py [hunk 3]
```diff
@@ -125,6 +152,34 @@ def process_is_target_same_as_suppressed_stimulus(self):
         qs = self._aggregate_query_by_breakdown(breakdown_query, experiments_subquery_by_breakdown)
         return qs
 
+    def process_is_cm_same_participants_as_task_optional_parameter(self):
+        experiments_subquery_by_breakdown = self.filtered_experiments.filter(
+            unconsciousness_measures__is_cm_same_participants_as_task=OuterRef("series_name")
```
Comment (dani-efr) [inline uncontrast_studies/processors/parameters_distribution_free_queries.py:157]: same question as before (plural vs. singular)

File: uncontrast_studies/processors/parameters_distribution_pie.py [hunk 4]
```diff
@@ -34,216 +30,209 @@ def process(self):
...
         return qs
 
-    def process_reporting(self):
+    def process_suppression_method(self):
         experiments_subquery_by_breakdown = (
-            AggregatedInterpretation.objects.filter(type=InterpretationsChoices.PRO)
-            .filter(experiment__in=self.experiments)
-            .filter(experiment__is_reporting=OuterRef("series_name"))
+            self.experiments
+            .filter(suppression_methods__type__name=OuterRef("series_name"))
```
Comment (dani-efr) [inline uncontrast_studies/processors/parameters_distribution_pie.py:78]: same question

**Review by dani-efr — APPROVED** [review]

File: uncontrast_studies/processors/experiments_comparison.py (hunk 1 above)
Comment (alonisser) [reply uncontrast_studies/processors/experiments_comparison.py:111]: Yes, it's a bit strange, but it seems so, as it's the "reverse relation"

File: uncontrast_studies/processors/experiments_comparison.py (hunk 2 above)
Comment (alonisser) [reply uncontrast_studies/processors/experiments_comparison.py:203]: copy paste i'll remove

**Pull request merged**